src/libraries/System.IO.FileSystem/tests/FileStream/ctor_options.cs

```diff
@@ -162,21 +162,23 @@ public void PreallocationSize(FileAccess access, FileMode mode)
         [InlineData(FileMode.CreateNew)]
         public void WhenDiskIsFullTheErrorMessageContainsAllDetails(FileMode mode)
         {
-            const long tooMuch = 1024L * 1024L * 1024L * 1024L; // 1 TB
-
+            const long tooMuch = 1024L * 1024L * 1024L * 1024L * 1024L * 1024L; // 1 Exbibyte .. we are assuming this is not available
             string filePath = GetTestFilePath();
 
-            IOException ex = Assert.Throws<IOException>(() => CreateFileStream(filePath, mode, FileAccess.Write, FileShare.None, bufferSize: 1, FileOptions.None, tooMuch));
-            Assert.Contains(filePath, ex.Message);
-            Assert.Contains(tooMuch.ToString(), ex.Message);
-
-            // ensure it was NOT created
-            bool exists = File.Exists(filePath);
-            if (exists)
+            try
             {
-                File.Delete(filePath);
+                // not using Assert.Throws because in the event of failure, we want to dispose, so the next test isn't affected
+                using FileStream fs = CreateFileStream(filePath, mode, FileAccess.Write, FileShare.None, bufferSize: 1, FileOptions.None, tooMuch);
+                Assert.Fail($"Expected to throw IOException, {fs.Length}");
```

Comment:
```suggestion
                Assert.Throws(() =>
                {
                    using FileStream fs = CreateFileStream(filePath, mode, FileAccess.Write, FileShare.None, bufferSize: 1, FileOptions.None, tooMuch);
                });
```

won't this work though?